gcloud/bigquery/job.py

```diff
@@ -919,3 +931,278 @@ def _scrub_local_properties(self, cleaned):
                 assert dest_remote['projectId'] == self.project
                 dataset = self._client.dataset(dest_remote['datasetId'])
                 self.destination_table = dataset.table(dest_remote['tableId'])
+
+
+class _SyncQueryConfiguration(object):
+    """User-settable configuration options for synchronous query jobs."""
+    # None -> use server default.
+    _default_dataset = None
+    _max_results = None
+    _timeout_ms = None
+    _preserve_nulls = None
+    _use_query_cache = None
+
+
+class RunSyncQueryJob(_BaseJob):
+    """Synchronous job: query tables.
+
+    :type query: string
+    :param query: SQL query string
+
+    :type client: :class:`gcloud.bigquery.client.Client`
+    :param client: A client which holds credentials and project configuration
+                   for the dataset (which requires a project).
+    """
+    def __init__(self, query, client):
+        super(RunSyncQueryJob, self).__init__(client)
+        self.query = query
+        self._configuration = _SyncQueryConfiguration()
+
+    @property
+    def cache_hit(self):
+        """Query results served from cache.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/v2/jobs/query#cacheHit
+
+        :rtype: boolean or ``NoneType``
+        :returns: True if the query results were served from cache (None
+                  until set by the server).
+        """
+        return self._properties.get('cacheHit')
+
+    @property
+    def complete(self):
+        """Server completed query.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/v2/jobs/query#jobComplete
+
+        :rtype: boolean or ``NoneType``
+        :returns: True if the query completed on the server (None
+                  until set by the server).
+        """
+        return self._properties.get('jobComplete')
+
+    @property
+    def errors(self):
+        """Errors generated by the query.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/v2/jobs/query#errors
+
+        :rtype: list of mapping, or ``NoneType``
+        :returns: Mappings describing errors generated on the server (None
+                  until set by the server).
+        """
+        return self._properties.get('errors')
+
+    @property
+    def name(self):
+        """Job name, generated by the back-end.
```

Comment:
The API is inconsistent here:  for all other jobs, `name` is the URL fragment (e.g., `/projects/your-project-456/jobs/user-provided-name`), while `id` is a generated compound (`your-project-456:user-provided-name`).  Synchronous query jobs don't allow the user to set the name, and then return it in the resource as `jobId` (*not* wrapped in a `jobReference`).

I chose to make the synchronous query version in the client match the others, rather than the API.